tools/king/king.xml

```diff
@@ -1,15 +1,77 @@
-
 <tool id="king" name="KING" version="2.2.4" python_template_version="3.5">
     <requirements>
-        <requirement type="package">king</requirement>
+	<requirement type="package">king</requirement>
     </requirements>
     <command detect_errors="exit_code"><![CDATA[
-    king -b $genotype --fam $family --bim $map --kinship > $kingoutput
+    king -b $genotype --fam $family --bim $map
+
+    #if $related
+    --related
+    #end if
+
+    #if $duplicate
+    --duplicate
+    #end if
+
+    #if $kinship
+    --kinship
+    #end if
+
+    #if $ibdseg
+    --ibdseg
+    #end if
+
+    #if $ibs
+    --ibs
+    #end if
+
+    #if $homog
+    --homog
+    #end if
+
+    #if $degree
+    --degree $degree
+    #end if
+
+    #if $projection
+    --projection $projection
+    #end if
+
+    #if $unrelated
+    --unrelated
+    #end if
+
+    #if $build
+    --build
+    #end if
+
+    #if $cluster
+    --cluster
+    #end if
+
+    #if $rplot
+    --rplot
+    #end if
+
+    > $kingoutput
     ]]></command>
     <inputs>
 	<param name="genotype" type="data" format="bed" help="Binary genotype File" />
 	<param name="family" type="data" format="txt" help="Family File" />
 	<param name="map" type="data" format="tabular" help="Map File" />
+	<!-- Beginning of the optional paramters -->
+	<param name="related" type="boolean" checked="false" optional="true" label="Relatedness" />
```

Comment:
`<param name="related" type="boolean" checked="false" truevalue="--related" falsevalue="" label="Relatedness" />`

Note that the `optional` tag does not make sense for booleans, because they are either on or off.

Another thing to note is that if the name of the parameter is the same as the command line flag, you can actually just do is use the [argument](https://docs.galaxyproject.org/en/master/dev/schema.html#id24) tag:

`<param argument="--related" type="boolean" checked="false" label="Relatedness" />`

this can then be referenced in the cheetah code in the `<command>` section as `$related`

Reply:
By using the argument tag, how I can control the test cases? I've tried different options but the value ends up being "false" rather than "--relatedness".